app/addons/documents/mango/components/ExecutionStats.js

```diff
@@ -0,0 +1,127 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+import React from 'react';
+import { Popover, OverlayTrigger } from 'react-bootstrap';
+
+const TOO_MANY_DOCS_SCANNED_WARNING = "The number of documents examined is high in proportion to the number of results returned. Consider adding an index to improve this.";
+
+export default class ExecutionStats extends React.Component {
+  constructor (props) {
+    super(props);
+  }
+
+  humanizeDuration(milliseconds) {
```

Comment:
We use moment.js. There should be a helper function to help with this, that might make this simpler

Reply:
aha - I didn't realise that and didn't want to pull in the dependency. If it's there, I'll use it.

Reply:
looking at it, the moment.js `humanize()` function rounds up milliseconds to seconds, so I don't think it's the right thing for this.

Reply:
ok that makes sense